### DISCUSSION TITLE
Approve members and assign roles in a single step

### DISCUSSION THREAD

**Discussion by ricklecoat** [ExpressionEngine/ExpressionEngine#3004] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Suppose a site has registrations set to ‘manual’. Somebody registers on the front end and their membership is placed in ‘Pending’. At this point a site administrator can approve it, which triggers a ‘Your account is active’ email to the waiting registree.  However, the site admin also then needs to **set the new member account to the correct primary role**, since the role allocated to new members by default may not be the correct one. Different roles can see different content, so if the admin is slow about setting the primary role (which can take some time, even if they're on the ball, as it requires several clicks and a password verification) and if the user is quick about logging in after getting their email, there is a chance that the user might see content that was not meant for them.

One way to avoid this is to set the role before approving the membership request — moving the member out of Pending and into a role with greater permissions seems to implicitly ‘approve’ the member. **However this method doesn't trigger the ‘your account is active’ email**.

**Describe the solution you'd like**
What would be very helpful would be if either:

1. Changing a role from Pending to some other role (other than Guest or Banned) triggered the approval notification email, or
2. Explicitly approving a member (clicking the green tick in ‘Pending Activation’ or the general ‘Member list’) provided a way to choose the members allocated role(s) as part of that same step, only sending the acceptance email once the roles have been set.

**Describe alternatives you've considered**
Other than the method of approving the member by changing their primary role away from Pending (which fails to send the email to the waiting registree and is therefore not a solution) I have no alternative ideas.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
In the recent versions of ExpressionEngine, it is possible to use `[primary_role](https://docs.expressionengine.com/latest/member/registration.html#primary_role)` with the member registration tag, overriding the default setting

What was still missing however is a way to change the pending role while the member still has not been activated. This is getting added in this PR: https://github.com/ExpressionEngine/ExpressionEngine/pull/4528

Clicking "activate" checkbox will approve the member into selected pending role and trigger activation email
